### PR TITLE
fix(files): Reset context menu position on close

### LINKED ISSUE #46934: [Bug]: Context menu breaks when clicking on "..." while still open from right click
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

On daily:

https://github.com/user-attachments/assets/0661a40e-629a-4c0d-9e5a-ed611f6d27f3

### Steps to reproduce

1. Open context menu with right click
2. Without closing it, click on "..." of a file

### PR DESCRIPTION
**Pull request by susnux**
* Resolves https://github.com/nextcloud/server/issues/46934

## Summary

Ensure that after a right-click the context menu position is reset, so that pressing the "actions"-menu button it is displayed on the correct location.

### Screen recording

[Bildschirmaufnahme_20240909_153456.webm](https://github.com/user-attachments/assets/f4122958-b4ac-4ea1-844e-8e4f50e9bdb8)

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by szaimen — APPROVED** [review]
LGTM from the video

**Review by marcelklehr — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Comment by marcelklehr** [comment]
Conflicts :/

**Comment by provokateurin** [comment]
@susnux I think this broke groupfolders cypress with the following error message: `visit/apps/files
 (uncaught exception)TypeError: Cannot read properties of null (reading 'style')`
It started failing at some point yesterday and this is the only matching commit for that time frame that also references `style`.

**Comment by provokateurin** [comment]
See https://github.com/nextcloud/groupfolders/actions/runs/10796406217/job/29947702773?pr=3201 for example.

File: apps/files/src/components/FileEntryMixin.ts [hunk 1]
```diff
@@ -220,6 +220,23 @@ export default defineComponent({
...
+				// ref: https://github.com/nextcloud-libraries/nextcloud-vue/pull/6065
+				window.setTimeout(() => {
+					if (this.openedMenu) {
+						// was reopened while the animation run
+						return
+					}
+					// Reset any right menu position potentially set
+					const root = this.$el?.closest('main.app-content') as HTMLElement
+					root.style.removeProperty('--mouse-pos-x')
+					root.style.removeProperty('--mouse-pos-y')
```
Comment (provokateurin) [inline apps/files/src/components/FileEntryMixin.ts:236]: Not null aware

Comment (susnux) [reply apps/files/src/components/FileEntryMixin.ts:236]: How could this ever be null? The FileEntry is only mounted within the files app, which is always inside `NcAppContent` -> `<main class="app-content">`

Comment (provokateurin) [reply apps/files/src/components/FileEntryMixin.ts:236]: No idea :woman_shrugging: 
But this.$el is nullable so anything down the chain can also be null.

Comment (susnux) [reply apps/files/src/components/FileEntryMixin.ts:236]: > But this.$el is nullable so anything down the chain can also be null.

yes if the component is not mounted at all

Comment (susnux) [reply apps/files/src/components/FileEntryMixin.ts:236]: lets fix it nevertheless :)

**Pull request merged**